smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptSettings.java

```diff
@@ -65,6 +66,7 @@ public final class TypeScriptSettings {
     private boolean isPrivate;
     private ArtifactType artifactType = ArtifactType.CLIENT;
     private boolean disableDefaultValidation = false;
+    private boolean backwardCompatibleRequiredMember = true;
```

Comment:
Can this be converted to an enum so that we can add additional modes in the future? I think the mode you're adding to remove `| undefined` could be `strict` or something similar.

There's an enum in NullableIndex as an example that has allowed for adding additional check modes as it has evolved: https://github.com/awslabs/smithy/blob/8beb77af9e9dd1ca8cdb822f3a99da5ecff24321/smithy-model/src/main/java/software/amazon/smithy/model/knowledge/NullableIndex.java#L54-L147